framework/ChiMesh/LogicalVolume/SurfaceMeshLogicalVolume.cc

```diff
@@ -0,0 +1,213 @@
+#include "SurfaceMeshLogicalVolume.h"
+
+#include "ChiMesh/chi_mesh.h"
+#include "ChiMesh/SurfaceMesh/chi_surfacemesh.h"
+#include "ChiMesh/Raytrace/raytracing.h"
+
+#include "ChiObject/object_maker.h"
+
+#include <utility>
+
+namespace chi_mesh
+{
+
+RegisterChiObject(chi_mesh, SurfaceMeshLogicalVolume);
+
+chi_objects::InputParameters SurfaceMeshLogicalVolume::GetInputParameters()
+{
+  chi_objects::InputParameters params = LogicalVolume::GetInputParameters();
+
+  // clang-format off
+  params.SetGeneralDescription(
+  "\\defgroup chi_mesh__SurfaceMeshLogicalVolume SurfaceMeshLogicalVolume\n"
+  "\\ingroup LuaLogicVolumes");
+  // clang-format on
+
+  params.AddRequiredParameter<size_t>(
+    "surface_mesh_handle",
+    "Handle to a surface mesh that will represent this object");
+
+  return params;
+}
+
+SurfaceMeshLogicalVolume::SurfaceMeshLogicalVolume(
+  const chi_objects::InputParameters& params)
+  : LogicalVolume(params),
+    surf_mesh(chi::GetStackItemPtrAsType<chi_mesh::SurfaceMesh>(
+      chi::object_stack,
+      params.GetParamValue<size_t>("surface_mesh_handle"),
+      __FUNCTION__)),
+    xbounds_({1.0e6, -1.0e6}),
+    ybounds_({1.0e6, -1.0e6}),
+    zbounds_({1.0e6, -1.0e6})
```

Comment:
Does this high, then low specification make sense? Usually you see low, then high (i.e. increasing order).

Reply:
The way this has been recoded these values aren't used, but it still requires something. In that sense:
`xmin,xmax = {-12, +12}`
first value `5`
`xmin = min(5, -12) = -12` and
`xmax = max(5, +12) = +12`. Clearly wrong.

Versus the other way around:
`xmin, xmax = {+12, -12}`
first value `5`
`xmin = min(5, +12) = 5` and
`xmax = max(5, -12) = 5`
works.